samples/seldon/mnist_tf.py

```diff
@@ -0,0 +1,222 @@
+# limitations under the License.
+
+from kubernetes import client as k8s_client
+import kfp.dsl as dsl
+import json
+
+
+@dsl.pipeline(
+    name="Seldon MNIST TF",
+    description="Example of training and serving seldon MNIST TF model. Requires docker secret as per kubeflow/example-seldon."
+)
+
+#Example derived from https://github.com/kubeflow/example-seldon
+#push access needed to chosen docker repo - see note below on secret
+def mnist_tf(docker_secret='docker-config',
+             training_repo='https://github.com/kubeflow/example-seldon.git',
+             training_branch='master',
+             training_files='./example-seldon/models/tf_mnist/train/*',
+             docker_repo_training='seldonio/deepmnistclassifier_trainer',
+             docker_tag_training='0.3',
+             serving_repo='https://github.com/kubeflow/example-seldon.git',
+             serving_branch='master',
+             serving_files='./example-seldon/models/tf_mnist/runtime/*',
+             docker_repo_serving='seldonio/deepmnistclassifier_runtime',
+             docker_tag_serving='0.3'):
+
+#will be pushing image so need docker secret
+#create local with `kubectl create secret generic docker-config --from-file=config.json=${DOCKERHOME}/config.json --type=kubernetes.io/config`
+    secret = k8s_client.V1Volume(
+        name="docker-config-secret",
+        secret=k8s_client.V1SecretVolumeSource(secret_name=docker_secret)
+    )
+
+#use volume for storing model
+    modelvolop = dsl.VolumeOp(
+        name="modelpvc",
+        resource_name="modelpvc",
+        size="50Mi",
+        modes=dsl.VOLUME_MODE_RWO
+    )
+#and another as working directory between steps
+    wkdirop = dsl.VolumeOp(
+        name="wkdirpvc",
+        resource_name="wkdirpvc",
+        size="50Mi",
+        modes=dsl.VOLUME_MODE_RWO
+    )
+
+#clone the training code and move to workspace dir as kaniko (next step) expects that
+    clone = dsl.ContainerOp(
+        name="clone",
+        image="alpine/git:latest",
+        command=["sh", "-c"],
+        arguments=["git clone --depth 1 --branch "+str(training_branch)+" "+str(training_repo)+"; cp "+str(training_files)+" /workspace; ls /workspace/;"],
+        pvolumes={"/workspace": wkdirop.volume}
+    )
+
+#build and push image for training
+    build = dsl.ContainerOp(
+        name="build",
+        image="gcr.io/kaniko-project/executor:latest",
+        arguments=["--dockerfile","Dockerfile","--destination",str(docker_repo_training)+":"+str(docker_tag_training)],
+        pvolumes={"/workspace": clone.pvolume,"/root/.docker/": secret}
+    )
+
+    tfjobjson = """
+{
+	"apiVersion": "kubeflow.org/v1beta1",
+	"kind": "TFJob",
+	"metadata": {
+		"name": "mnist-train-{{workflow.uid}}"
+	},
+	"spec": {
+		"tfReplicaSpecs": {
+			"Worker": {
+				"replicas": 1,
+				"template": {
+					"spec": {
+						"containers": [
+							{
+								"image": "{{workflow.parameters.docker-repo-training}}:{{workflow.parameters.docker-tag-training}}",
```

Comment:
Yeah substituting in here is cool - thanks for pointing it out. I am going with Template as the $ format for variables works better here than curly-braces since the json contains a lot of curly-braces.